pkg/query-service/app/clickhouseReader/options.go

```diff
@@ -108,6 +110,11 @@ func defaultConnector(cfg *namespaceConfig) (clickhouse.Conn, error) {
 		}
 		options.Auth = auth
 	}
+	if strings.ToLower(dsnURL.Query().Get("secure")) == "true" {
```

Comment:
I suggest using [`strconv.ParseBool`](https://pkg.go.dev/strconv#ParseBool) for this since it handles the casing and matches more than just `true`

For reference, [here is how these options are parsed](https://github.com/ClickHouse/clickhouse-go/blob/a6e001eb87398dec2062dfbe02f27b9168c51fd3/clickhouse_options.go#L242) in the `fromDSN` function from `clickhouse-go`